src/S3/PostObjectV4.php

```diff
@@ -0,0 +1,209 @@
+     *                                      default: 1 hour vaild peroid.
+     */
+    public function __construct(
+        S3ClientInterface $client,
+        $bucket,
+        array $formInputs,
+        array $options = [],
+        $expiration = NULL
+    ) {
+        $this->ldt = gmdate(self::ISO8601_BASIC);
+        $this->sdt = substr($this->ldt, 0, 8);
+
+        $this->client = $client;
+        $this->bucket = $bucket;
+
+        // setup form attributes
+        $this->formAttributes = [
+            'action'  => $this->generateUri(),
+            'method'  => 'POST',
+            'enctype' => 'multipart/form-data'
+        ];
+
+        // setup basic policy
+        $expiration = is_null($expiration)
+                ? gmdate('Y-m-d\TH:i:s\Z', strtotime('+1 hours'))
+                : TimestampShape::format($expiration, 'iso8601');
+
+        $policy = [
+            'expiration' => $expiration,
+            'conditions' => [],
+        ];
+        foreach ($options as $option) {
+            $policy['conditions'][] = $option;
+        }
+
+        $this->formInputs = $formInputs + ['key' => '${filename}'];
+        // add v4 conditions for policy and formInputs
+        foreach ($this->getV4Conditions() as $key => $value) {
+            $policy['conditions'][] = [$key => $value];
+            $this->formInputs += [$key => $value];
+        }
+        $this->jsonPolicy = json_encode($policy);
+
+        // add signature
+        $this->formInputs += $this->getPolicyAndSignature(
+            $this->client->getCredentials()->wait()
+        );
+    }
+
+    /**
+     * Gets the S3 client.
+     *
+     * @return S3ClientInterface
+     */
+    public function getClient()
+    {
+        return $this->client;
+    }
+
+    /**
+     * Gets the bucket name.
+     *
+     * @return string
+     */
+    public function getBucket()
+    {
+        return $this->bucket;
+    }
+
+    /**
+     * Gets the form attributes as an array.
+     *
+     * @return array
+     */
+    public function getFormAttributes()
+    {
+        return $this->formAttributes;
+    }
+
+    /**
+     * Set a form attribute.
+     *
+     * @param string $attribute Form attribute to set.
+     * @param string $value     Value to set.
+     */
+    public function setFormAttribute($attribute, $value)
+    {
+        $this->formAttributes[$attribute] = $value;
+    }
+
+    /**
+     * Gets the form inputs as an array.
+     *
+     * @return array
+     */
+    public function getFormInputs()
+    {
+        return $this->formInputs;
+    }
+
+    /**
+     * Set a form input.
+     *
+     * @param string $field Field name to set
+     * @param string $value Value to set.
+     */
+    public function setFormInput($field, $value)
+    {
+        $this->formInputs[$field] = $value;
+    }
+
+    /**
+     * Gets the raw JSON policy.
+     *
+     * @return string
+     */
+    public function getJsonPolicy()
```

Comment:
I'm not sure this should be exposed as a public method on a V4 signer class. The policy isn't final until it's signed.